nifi-nar-bundles/nifi-soap-bundle/nifi-soap-processors/src/main/java/org/apache/nifi/processors/soap/GetSOAP.java

```diff
@@ -46,34 +51,30 @@
 import org.apache.nifi.processor.ProcessorInitializationContext;
 import org.apache.nifi.processor.Relationship;
 import org.apache.nifi.processor.exception.ProcessException;
-import org.apache.nifi.processor.io.OutputStreamCallback;
 import org.apache.nifi.processor.util.StandardValidators;
-import org.apache.nifi.util.StopWatch;
+
 import java.io.IOException;
-import java.io.OutputStream;
 import java.util.*;
 
-import static org.apache.axis2.namespace.Constants.URI_SOAP12_ENV;
-
-
 @SupportsBatching
-@InputRequirement(InputRequirement.Requirement.INPUT_FORBIDDEN)
+@InputRequirement(InputRequirement.Requirement.INPUT_REQUIRED)
 @Tags({"SOAP", "Get", "Ingest", "Ingress"})
-@CapabilityDescription("Execute provided request against the SOAP endpoint. The result will be left in it's orginal form. " +
+@CapabilityDescription(
+        "Execute provided request against the SOAP endpoint. The result will be left in it's orginal form. " +
         "This processor can be scheduled to run on a timer, or cron expression, using the standard scheduling methods, " +
         "or it can be triggered by an incoming FlowFile. If it is triggered by an incoming FlowFile, then attributes of " +
         "that FlowFile will be available when evaluating the executing the SOAP request.")
 @WritesAttribute(attribute = "mime.type", description = "Sets mime type to text/xml")
```

Comment:
By definition, SOAP uses XML but if you want to change it to be dynamic, that is fine by me. The code that reads the response will probably break if it isn't XML.